### PR TITLE
Fix issue with extending post types and reserved keywords.

### PR DESCRIPTION
**Pull request by JoryHogeveen**
## Description

Fix an issue related to #5428 (reserved keywords). When extending CPT or taxonomies it gave errors on (for example) `page` post type since that is also a reserved keyword.

## Types of changes

I've brought the `create_extend` parameter into `save_pod`. If that is set and is set to `extend` it will validate against existing CPT/Tax objects. If that name exists the name will still be valid.

## ChangeLog

No need for an extra changelog.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: classes/PodsAPI.php [hunk 1]
```diff
@@ -1753,12 +1760,28 @@ public function save_pod( $params, $sanitized = false, $db = true ) {
...
 		} else {
+			// New pod (create).
+
+			if (
+				in_array( $params->name, pods_reserved_keywords(), true )
+				&& in_array( pods_v( 'type', $params ), array( 'post_type', 'taxonomy' ), true )
+			) {
+				$valid_name = false;
+
+				if ( $extend ) {
```
Comment (sc0ttkclark) [inline classes/PodsAPI.php]:
```suggestion
				if ( ! $extend ) {
```

@JoryHogeveen It seems like we only want to validate on creating new ones, extending existing ones is not under our control and the plugin/theme that created the post type / taxonomy would have to mitigate their own risks.

Comment (JoryHogeveen) [reply classes/PodsAPI.php]: Correct, but this code works the other way around.
If rejects all reserved keywords (see line 1767).
When if finds a reserved keyword THEN it checks if `$extend` is true. If it's true then existing CPT/Tax are allowed.

Comment (JoryHogeveen) [reply classes/PodsAPI.php]: I'll add inline documentation

**Comment by JoryHogeveen** [comment]
With #5453 merged the errors are shown again.

- Tested adding Pods with reserved keywords. Correctly returns error message.
- Tested extending post types that are reserved keywords (page). Extending works.
- Tested creating exiting post type. Returns error message.
- Tested with WP_DEBUG on and off. Both return error messages (#5453) 

IMO, Q&A is passed.

**Pull request merged**